src/infra/core/src/repos/dataset_factory_impl.rs

```diff
@@ -328,10 +328,20 @@ impl DatasetFactory for DatasetFactoryImpl {
                 let ds = Self::get_local_fs(layout, self.event_bus.clone());
                 Ok(Arc::new(ds) as Arc<dyn Dataset>)
             }
-            "http" | "https" | "odf+http" | "odf+https" => {
+            "http" | "https" => {
                 let ds = Self::get_http(url, self.build_header_map(url), self.event_bus.clone());
                 Ok(Arc::new(ds))
             }
+            "odf+http" | "odf+https" => {
+                // TODO: PERF: Consider what speedups are possible in smart protocol
+                let http_url = Url::parse(url.as_str().strip_prefix("odf+").unwrap()).unwrap();
+                let ds = Self::get_http(
+                    &http_url,
+                    self.build_header_map(&http_url),
```

Comment:
Per #521 `kamu pull repo/account/dataset` was resulting in using SimpleTP with `odf+http` protocol and the `build_header_map` here was crashing when accessing `.kamutokenstore` as it expected raw http/https URL. So I'm stripping off `odf+` schema here.